service/src/LevelZero.cpp

```diff
@@ -426,6 +510,45 @@ namespace geopm
         return m_devices.at(l0_device_idx).subdevice.engine_domain.at(l0_domain).size();
     }
 
+    int LevelZeroImp::performance_domain_count(int geopm_domain, unsigned int l0_device_idx, int l0_domain) const
+    {
+        int result = 0;
+        if (geopm_domain == GEOPM_DOMAIN_GPU) {
+            result = m_devices.at(l0_device_idx).perf_domain.size();
+        }
+        else if (geopm_domain == GEOPM_DOMAIN_GPU_CHIP) {
+            result = m_devices.at(l0_device_idx).subdevice.perf_domain.at(l0_domain).size();
+        }
+        return result;
```

Comment:
This will silently proceed when an invalid domain is provided and return a default result of 0. Looking at it from an interface consistency point of view, there are other methods that throw in this situation, instead: the preexisting ``num_gpu()`` method and the newly added ``performance_factor_control()``. What criteria do you use to determine which behavior applies? If it's not very clear and there's potential for confusion, consider addressing this consistency issue.

Reply:
The _domain_count functions are intended to be used to check if it is sensible to call into any of the APIs around the associated feature (power, perf_factor, frequency, etc), hence they return 0.  The LevelZeroDevicePool::check_domain_exists will throw an error in this case.  

The control functions have additional checking around them as a safety precaution for conditions that are specifically not supported to avoid ending up in a case where something is changed on the system due to bad write.